galsim/gsobject.py

```diff
@@ -1412,84 +1736,84 @@ def drawKImage(self, re=None, im=None, nx=None, ny=None, bounds=None, scale=None
-                            A provided `im` must match the size and scale of `re`.
-                            If `im` is None, then `re` must also be None. [default: None]
+        @param nx           If provided and `image` is None, use to set the x-direction size of the
+                            image.  Must be accompanied by `ny`.
+        @param ny           If provided and `image` is None, use to set the y-direction size of the
+                            image.  Must be accompanied by `nx`.
+        @param bounds       If provided and `image` is None, use to set the bounds of the image.
         @param scale        If provided, use this as the pixel scale, dk, for the images.
-                            If `scale` is None and `re` and `im` are given, then take the provided
+                            If `scale` is None and `image` is given, then take the provided
                             images' pixel scale (which must be equal).
-                            If `scale` is None and `re` and `im` are None, then use the Nyquist
-                            scale.
-                            If `scale <= 0` (regardless of `re`, `im`), then use the Nyquist scale.
+                            If `scale` is None and `image` is None, then use the Nyquist scale.
+                            If `scale <= 0` (regardless of `image`), then use the Nyquist scale.
                             [default: None]
-        @param dtype        The data type to use for automatically constructed images.  Only
-                            valid if `re` and `im` are None. [default: None, which means to
-                            use numpy.float32]
-        @param gain         The number of photons per ADU ("analog to digital units", the units of
-                            the numbers output from a CCD).  [default: 1.]
-        @param wmult        A multiplicative factor by which to enlarge (in each direction) the
-                            size of the image, if you are having drawKImage() automatically
-                            construct the images for you.  [default: 1]
         @param add_to_image Whether to add to the existing images rather than clear out
                             anything in the image before drawing.
-                            Note: This requires that `re` and `im` be provided and that they have
-                            defined bounds. [default: False]
+                            Note: This requires that `image` be provided and that it has defined
+                            bounds. [default: False]
 
-        @returns the tuple of Image instances, `(re, im)` (created if necessary)
+        @returns an ImageC instance (created if necessary)
         """
-        # Check for obsolete dk parameter
+        if isinstance(image,galsim.Image) and isinstance(nx,galsim.Image):
+            # If the user calls drawK(re,im), then give the proper deprecation below.
```

Comment:
This will give a warning, just not in this code block.  It sets re, im, which a few lines down will call depr, which raises a warning.